pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -148,48 +148,31 @@ def get_epoch_log_metrics(self, *_, **__) -> List[Dict]:
     def get_forked_metrics(self, *_, **__) -> List[Dict]:
         return self.get_epoch_from_func_name("get_forked_metrics")
 
-    @staticmethod
-    def _append_to_structure(primary_dict, opt_idx, batch_idx, result) -> None:
-        primary_dict.setdefault(opt_idx, {})
-        primary_dict[opt_idx].setdefault(batch_idx, [])
-        primary_dict[opt_idx][batch_idx].append(result)
-
-    def append(self, result, dataloader_idx: Optional[int] = None, extra_info: Optional[dict] = None) -> None:
-        assert isinstance(result, Result)
-        if dataloader_idx is None:
-            dataloader_idx = 0
-
-        if extra_info is None:
-            extra_info = {}
-
-        # [dataloader_idx][optimizer_idx][training_step_idx] is a list
-        if len(extra_info) > 0:
-            self._internal_type = ResultStoreType.INSIDE_BATCH_TRAIN_LOOP
-            # initialize dictionary
+    def append(self, result: Result, info: Dict) -> None:
+        dataloader_idx = info["dataloader_idx"]
+        self._internal_type = info["type"]
+        opt_idx = info["opt_idx"]
+
+        if self._internal_type == ResultStoreType.INSIDE_BATCH_TRAIN_LOOP:
+            # [dataloader_idx][optimizer_idx][training_step_idx] is a list
             if dataloader_idx not in self._internals:
                 self._internals[dataloader_idx] = {}
                 self._internals_reduced[dataloader_idx] = defaultdict(dict)
                 self._latest_ref[dataloader_idx] = {}
 
-            # extract infos
-            opt_idx = extra_info["opt_idx"]
-            batch_idx = extra_info["batch_idx"]
-
-            self._append_to_structure(self._internals[dataloader_idx], opt_idx, batch_idx, result)
-
-            self._latest_ref[dataloader_idx][opt_idx] = result
-
-        # [dataloader_idx] is a list
+            batch_idx = info["batch_idx"]
+            self._internals[dataloader_idx].setdefault(opt_idx, {})
+            self._internals[dataloader_idx][opt_idx].setdefault(batch_idx, [])
+            self._internals[dataloader_idx][opt_idx][batch_idx].append(result)
         else:
-            self._internal_type = ResultStoreType.OUTSIDE_BATCH_TRAIN_LOOP
             self._internals.setdefault(dataloader_idx, [])
             self._internals[dataloader_idx].append(result)
 
             if dataloader_idx not in self._latest_ref:
                 self._latest_ref[dataloader_idx] = {}
                 self._latest_ref[dataloader_idx][0] = {}
```

Comment:
Might have the same trouble there no ? `self._latest_ref[dataloader_idx][opt_idx]`

Reply:
What do you mean?

Reply:
```suggestion
                self._latest_ref[dataloader_idx][opt_idx] = {}
```

Reply:
I didn't try it,  but worth to check.

Reply:
Done